src/ui/widget.cpp

```diff
@@ -737,6 +737,10 @@ void Widget::saveLayout()
 void Widget::setDecorativeWidgetBounds()
 {
   onSetDecorativeWidgetBounds();
+  // Adjust the bounds of the window title label if a mini-button
+  // overlaps it.
+  if (parent() && parent()->window())
+    parent()->window()->adjustTitleLabelBounds();
```

Comment:
I think we can avoid creating a new window method (adjustTitleLabelBounds) if we override the onSetDecorativeWidgetBounds() in the window class and put the logic there after calling Widget::onSetDecorativeWidgetBounds().

Reply:
You're right! I followed your advice, but I assigned 'onSetDecorativeWidgetBounds' to 'Window TitleLabel' instead of 'Window'.